File: Dataset/Leetcode/train/2/107.java
class Solution {
    public ListNode XXX(ListNode l1, ListNode l2) {
        int carry = 0;
        ListNode res = new ListNode(0);
        ListNode cur = res;
        while(l1 != null || l2 != null) {
            if(l1 != null && l2 != null) {
                int temp = l1.val + l2.val + carry;
                cur.next = new ListNode(temp % 10);
                cur = cur.next;
                carry = temp / 10;
                l1 = l1.next;
                l2 = l2.next;
            }
            else if(l2 != null) {
                int temp = l2.val + carry;
                cur.next = new ListNode(temp % 10);
                cur = cur.next;
                carry = temp / 10;
                l2 = l2.next;
            }
            else if(l1 != null) {
                int temp = l1.val + carry;
                cur.next = new ListNode(temp % 10);
                cur = cur.next;
                carry = temp / 10;
                l1 = l1.next;
            }
        }
        if(carry > 0) {
            cur.next = new ListNode(carry);
        }
        return res.next;
    }
}

File: Dataset/Leetcode/train/69/107.cpp
class Solution {
public:
    int XXX(int x) {
        for(int i=0;i<=46339;i++){
            if(i*i==x){
                return i;
            }
            else if(i*i<x && (i+1)*(i+1)>x){
                return i;
            }
        }
        //2的32次方-1 再开方是46340.9 上面46340的平方都比x小 说明只能是取整就是46340 小数省略
        return 46340;
    }
};

